### PR TITLE
Mali/wall 3591

### PR DESCRIPTION
**Pull request by maliheh-deriv**
# Description : 
expand the test to cover and catch reported issue in production.

# cards 
https://app.clickup.com/t/20696747/WALL-3591

### CODE REVIEW

File: cypress/e2e/full/redirection.cy.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    cy.c_login()
+  })
+
+  it('should remain logged in after redirection to another platform', () => {
+    cy.findByTestId('dt_platform_switcher').click()
+    cy.get('a[href="/bot"]').click()
+    cy.get('.dc-dialog__footer > .dc-btn--secondary').click()
+    cy.get('#id-charts').click()
+    cy.get('.cq-top-ui-widgets > .ciq-menu > .cq-menu-btn').click()
+    cy.get('.sc-mcd__item--1HZ10V > .sc-mcd__item__name').click()
```
Comment (abuzar-deriv) [inline cypress/e2e/full/redirection.cy.js:13]: Can you please use playwright generator to find testing library supported commands?

File: cypress/e2e/full/redirection.cy.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    cy.findByTestId('dt_platform_switcher').click()
+    cy.get('a[href="/"]').click()
+    // Verify the chart loaded 
+    cy.xpath('//*[contains(@class,"smartcharts") and contains(@class, "loading")]').should('exist')
+    // Verify account remained logged in
+    cy.get('.dc-btn').should('exist')
```
Comment (abuzar-deriv) [inline cypress/e2e/full/redirection.cy.js:19]: Same as above for these. We will only default to cypress locators if we can't testing library commands. Xpath is a no go in almost all cases.

File: cypress/e2e/full/redirection.cy.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import '@testing-library/cypress/add-commands'
+describe('WALL-3591 -production bug WALL-3588', () => {
```
Comment (abuzar-deriv) [inline cypress/e2e/full/redirection.cy.js:2]: Can you please update the description to what the test is doing?

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]

File: cypress/e2e/full/redirection.cy.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import '@testing-library/cypress/add-commands'
+describe('WALL-3591 - Verify changing platforms and markets in Dtrader', () => {
+  beforeEach(() => {
+    cy.c_login()
+  })
+
+  it('should remain logged in after redirection to another platform', () => {
+    cy.findByTestId('dt_platform_switcher').click()
+    cy.get('a[href="/bot"]').click()
```
Comment (meeyun-deriv) [inline cypress/e2e/full/redirection.cy.js:9]: it is possible to use testing library for selecting different plaftform, e.g. for DBOT:
```
getByTestId('dt_trading-app-card_real_deriv-bot').getByRole('button', { name: 'Open' }).click();
```

File: cypress/e2e/full/redirection.cy.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    cy.findByRole('button', { name: 'Skip' }).click()
+    cy.findByText('Charts').click()
+    cy.get('.cq-symbol-select-btn').click()
+    cy.get('#dbot').findByText('Volatility 10 (1s) Index').click()
+    cy.findByTestId('dt_platform_switcher').click()
+    cy.get('a[href="/"]').click()
```
Comment (meeyun-deriv) [inline cypress/e2e/full/redirection.cy.js:15]: similar to comment above

**Review by meeyun-deriv — CHANGES_REQUESTED** [review]

**Review by abuzar-deriv — APPROVED** [review]

**Comment by maliheh-deriv** [comment]
created new PR https://github.com/deriv-com/e2e-deriv-app/pull/240

**Pull request closed without merging**